File: gluecodium/src/test/resources/smoke/comments/output/swift/smoke/CommentsMarkdown.swift
//
//
import Foundation
/// First line.
/// Second line.
///
/// Another paragraph. **bold** and *italic* and `code`.
///
/// > blockquote
///
/// # Heading one
///
/// ## Heading two
///
/// ### Heading three
///
/// Unordered list:
/// - A
/// - B
///
/// Ordered list:
/// 1. foo
/// 2. bar
///
/// ---
///
/// [title](https://www.markdownguide.org/cheat-sheet/)
public class CommentsMarkdown {
    let c_instance : _baseRef
    init(cCommentsMarkdown: _baseRef) {
        guard cCommentsMarkdown != 0 else {
            fatalError("Nullptr value is not supported for initializers")
        }
        c_instance = cCommentsMarkdown
    }
    deinit {
        smoke_CommentsMarkdown_remove_swift_object_from_wrapper_cache(c_instance)
        smoke_CommentsMarkdown_release_handle(c_instance)
    }
}
internal func getRef(_ ref: CommentsMarkdown?, owning: Bool = true) -> RefHolder {
    guard let c_handle = ref?.c_instance else {
        return RefHolder(0)
    }
    let handle_copy = smoke_CommentsMarkdown_copy_handle(c_handle)
    return owning
        ? RefHolder(ref: handle_copy, release: smoke_CommentsMarkdown_release_handle)
        : RefHolder(handle_copy)
}
extension CommentsMarkdown: NativeBase {
    /// :nodoc:
    var c_handle: _baseRef { return c_instance }
}
extension CommentsMarkdown: Hashable {
    /// :nodoc:
    public static func == (lhs: CommentsMarkdown, rhs: CommentsMarkdown) -> Bool {
        return lhs.c_handle == rhs.c_handle
    }
    /// :nodoc:
    public func hash(into hasher: inout Hasher) {
        hasher.combine(c_handle)
    }
}
internal func CommentsMarkdown_copyFromCType(_ handle: _baseRef) -> CommentsMarkdown {
    if let swift_pointer = smoke_CommentsMarkdown_get_swift_object_from_wrapper_cache(handle),
        let re_constructed = Unmanaged<AnyObject>.fromOpaque(swift_pointer).takeUnretainedValue() as? CommentsMarkdown {
        return re_constructed
    }
    let result = CommentsMarkdown(cCommentsMarkdown: smoke_CommentsMarkdown_copy_handle(handle))
    smoke_CommentsMarkdown_cache_swift_object_wrapper(handle, Unmanaged<AnyObject>.passUnretained(result).toOpaque())
    return result
}
internal func CommentsMarkdown_moveFromCType(_ handle: _baseRef) -> CommentsMarkdown {
    if let swift_pointer = smoke_CommentsMarkdown_get_swift_object_from_wrapper_cache(handle),
        let re_constructed = Unmanaged<AnyObject>.fromOpaque(swift_pointer).takeUnretainedValue() as? CommentsMarkdown {
        smoke_CommentsMarkdown_release_handle(handle)
        return re_constructed
    }
    let result = CommentsMarkdown(cCommentsMarkdown: handle)
    smoke_CommentsMarkdown_cache_swift_object_wrapper(handle, Unmanaged<AnyObject>.passUnretained(result).toOpaque())
    return result
}
internal func CommentsMarkdown_copyFromCType(_ handle: _baseRef) -> CommentsMarkdown? {
    guard handle != 0 else {
        return nil
    }
    return CommentsMarkdown_moveFromCType(handle) as CommentsMarkdown
}
internal func CommentsMarkdown_moveFromCType(_ handle: _baseRef) -> CommentsMarkdown? {
    guard handle != 0 else {
        return nil
    }
    return CommentsMarkdown_moveFromCType(handle) as CommentsMarkdown
}
internal func copyToCType(_ swiftClass: CommentsMarkdown) -> RefHolder {
    return getRef(swiftClass, owning: false)
}
internal func moveToCType(_ swiftClass: CommentsMarkdown) -> RefHolder {
    return getRef(swiftClass, owning: true)
}
internal func copyToCType(_ swiftClass: CommentsMarkdown?) -> RefHolder {
    return getRef(swiftClass, owning: false)
}
internal func moveToCType(_ swiftClass: CommentsMarkdown?) -> RefHolder {
    return getRef(swiftClass, owning: true)
}

File: gluecodium/src/test/resources/smoke/comments/output/dart/lib/src/smoke/comments_markdown.dart
import 'dart:ffi';
import 'package:library/src/_library_context.dart' as __lib;
import 'package:library/src/_native_base.dart' as __lib;
import 'package:library/src/_token_cache.dart' as __lib;
/// First line.
///
/// Second line.
///
/// Another paragraph. **bold** and *italic* and `code`.
///
/// > blockquote
///
/// # Heading one
///
/// ## Heading two
///
/// ### Heading three
///
/// Unordered list:
/// - A
/// - B
///
/// Ordered list:
/// 1. foo
/// 2. bar
///
/// ---
///
/// [title](https://www.markdownguide.org/cheat-sheet/)
abstract class CommentsMarkdown {
  /// @nodoc
  @Deprecated("Does nothing")
  void release();
}
// CommentsMarkdown "private" section, not exported.
final _smokeCommentsmarkdownRegisterFinalizer = __lib.catchArgumentError(() => __lib.nativeLibrary.lookupFunction<
    Void Function(Pointer<Void>, Int32, Handle),
    void Function(Pointer<Void>, int, Object)
  >('library_smoke_CommentsMarkdown_register_finalizer'));
final _smokeCommentsmarkdownCopyHandle = __lib.catchArgumentError(() => __lib.nativeLibrary.lookupFunction<
    Pointer<Void> Function(Pointer<Void>),
    Pointer<Void> Function(Pointer<Void>)
  >('library_smoke_CommentsMarkdown_copy_handle'));
final _smokeCommentsmarkdownReleaseHandle = __lib.catchArgumentError(() => __lib.nativeLibrary.lookupFunction<
    Void Function(Pointer<Void>),
    void Function(Pointer<Void>)
  >('library_smoke_CommentsMarkdown_release_handle'));
class CommentsMarkdown$Impl extends __lib.NativeBase implements CommentsMarkdown {
  CommentsMarkdown$Impl(Pointer<Void> handle) : super(handle);
  @override
  void release() {}
}
Pointer<Void> smokeCommentsmarkdownToFfi(CommentsMarkdown value) =>
  _smokeCommentsmarkdownCopyHandle((value as __lib.NativeBase).handle);
CommentsMarkdown smokeCommentsmarkdownFromFfi(Pointer<Void> handle) {
  final instance = __lib.getCachedInstance(handle);
  if (instance != null && instance is CommentsMarkdown) return instance;
  final _copiedHandle = _smokeCommentsmarkdownCopyHandle(handle);
  final result = CommentsMarkdown$Impl(_copiedHandle);
  __lib.cacheInstance(_copiedHandle, result);
  _smokeCommentsmarkdownRegisterFinalizer(_copiedHandle, __lib.LibraryContext.isolateId, result);
  return result;
}
void smokeCommentsmarkdownReleaseFfiHandle(Pointer<Void> handle) =>
  _smokeCommentsmarkdownReleaseHandle(handle);
Pointer<Void> smokeCommentsmarkdownToFfiNullable(CommentsMarkdown? value) =>
  value != null ? smokeCommentsmarkdownToFfi(value) : Pointer<Void>.fromAddress(0);
CommentsMarkdown? smokeCommentsmarkdownFromFfiNullable(Pointer<Void> handle) =>
  handle.address != 0 ? smokeCommentsmarkdownFromFfi(handle) : null;
void smokeCommentsmarkdownReleaseFfiHandleNullable(Pointer<Void> handle) =>
  _smokeCommentsmarkdownReleaseHandle(handle);
// End of CommentsMarkdown "private" section.
